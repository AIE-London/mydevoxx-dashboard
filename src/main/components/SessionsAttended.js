/**
 * Created by tsadler on 06/04/2017.
 */
import React from 'react';
import {Grid, Row, Col} from 'react-flexbox-grid';
import styled from 'styled-components';
import Card from './Card';
import Default from './defaultIcon.png';

const Container = styled.div`
  width: 400px;
  height: 400px;
`;

const StatMainHeader = styled.h1`
    text-align: left;
    font-size: 2em;
    font-family: helvetica;
`;

const CardHeader = styled.h2`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 1.75em;
  margin-bottom: 0;
  text-align: left;
`;

const CardSubhead = styled.h3`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 1.5em;
  margin-bottom: 0;
`;

const Image = styled.img`
    width: 50px;
    height: 50px;
    marginTop: 20px;
`;

class sessionsAttended extends React.Component {
    render() {
        return <Container>
            <Card>
                <Col xs={12}>
                    <Col xs={10}>
                        <Row start="xs">
                            <StatMainHeader>{this.props.mainHeader}</StatMainHeader>
                        </Row>
                    </Col>

                    <Row center="xs">
                        <Col xs={10}>
                            <Row start="xs" middle="xs">
                                <Col xs>
                                    <CardHeader>{this.props.title}</CardHeader>
                                    <p>{this.props.cardContent}</p>
                                </Col>
                                <Col xs={2}><Image src={Default}></Image></Col>
                            </Row>
                        </Col>
                    </Row>

                    <Row center="xs">
                        <Col xs={10}>
                            <Row start="xs" middle="xs">
                                <Col xs>
                                    <CardSubhead>{this.props.subHeader}</CardSubhead>
                                    <p>{this.props.name}</p>
                                </Col>
                            </Row>
                        </Col>
                    </Row>
                </Col>
            </Card>
        </Container>
    }
}
export default sessionsAttended;
