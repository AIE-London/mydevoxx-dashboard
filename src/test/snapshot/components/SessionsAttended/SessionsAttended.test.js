/**
 * Created by tsadler on 07/04/2017.
 */
import React from 'react';
import ReactDOM from 'react-dom';
import SessionsAttended from '../../../../main/components/SessionsAttended/';
import renderer from 'react-test-renderer';

test('sessionsAttended component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SessionsAttended title="Welcome to Devoxx" />, div);
    });
});

test('sessionsAttended component with title snapshot', () => {

    const tree = renderer.create(
        <SessionsAttended mainHeader="MySessions" title="Polymer - Web Standards"
                          cardContent ="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum eget libero mi. Ut vulputate leo in velit porta sollicitudin. Pellentesque vitae felis maximus, gravida mauris sed, bibendum turpis."
                          subHeader="Speaker:" name="Dan Cotton" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
