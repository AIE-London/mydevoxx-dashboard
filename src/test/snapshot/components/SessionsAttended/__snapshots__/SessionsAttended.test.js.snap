exports[`test sessionsAttended component with title snapshot 1`] = `
<div
  className="coDiMZ">
  <div
    className="bvnlUe row">
    <div
      className="col-xs-12">
      <div
        className="col-xs-10">
        <div
          className="row start-xs">
          <h1
            className="dEhMDz">
            MySessions
          </h1>
        </div>
      </div>
      <div
        className="row center-xs">
        <div
          className="col-xs-10">
          <div
            className="row start-xs middle-xs">
            <div
              className="col-xs">
              <h2
                className="gkGJEX">
                Polymer - Web Standards
              </h2>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum eget libero mi. Ut vulputate leo in velit porta sollicitudin. Pellentesque vitae felis maximus, gravida mauris sed, bibendum turpis.
              </p>
            </div>
            <div
              className="col-xs-2">
              <img
                className="gosEQ"
                src="defaultIcon.png" />
            </div>
          </div>
        </div>
      </div>
      <div
        className="row center-xs">
        <div
          className="col-xs-10">
          <div
            className="row start-xs middle-xs">
            <div
              className="col-xs">
              <h3
                className="jJURiy">
                Speaker:
              </h3>
              <p>
                Dan Cotton
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
